### PR TITLE
Or 1615 apply superchain config

### PR DESCRIPTION
**Pull request by boohyung**
Applied to support [SuperchainConfig](https://github.com/tokamak-network/tokamak-thanos/pull/171#issue-2321286188). Now we can pause and unpause contracts for withdrawal using SuperchainConfig. And we use superchainConfigGuardian instead of portalguardian in deploy config.
* add SuperchainConfig and update related L1 contracts
* add unit test for SuperchainConfig
* update deploy script and deploy config including upgrade SuperchainConfig

### CODE REVIEW

**Review by 0x6e616d — APPROVED** [review]
Thank you, approve first, I am updating the `devnet/main.py` file so you can ignore the failed ci.

**Comment by boohyung** [comment]
> Thank you, approve first, I am updating the `devnet/main.py` file so you can ignore the failed ci.

Thank you for your kindness, Nam. Let me check deploy script and more too.

File: packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol [hunk 1]
```diff
@@ -157,24 +149,26 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
...
-    function pause() external {
-        require(msg.sender == guardian, "OptimismPortal: only guardian can pause");
-        paused = true;
-        emit Paused(msg.sender);
+    /// @notice Getter function for the address of the guardian.
+    ///         Public getter is legacy and will be removed in the future. Use `SuperchainConfig.guardian()` instead.
+    /// @return Address of the guardian.
+    /// @custom:legacy
+    function guardian() public view returns (address) {
+        return superchainConfig.guardian();
```
Comment (0x6e616d) [inline packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol:165]: This function returns the zero address when running devnet script

Comment (boohyung) [reply packages/tokamak/contracts-bedrock/src/L1/OptimismPortal.sol:165]: Thanks for checking. Let me check it as well.

**Review by rlgns98kr — APPROVED** [review]
Thank you very much, Theo! ❤️🙇🏻‍♂️

**Pull request merged**